src/httpserver.cpp

```diff
@@ -639,6 +641,38 @@ HTTPRequest::RequestMethod HTTPRequest::GetRequestMethod() const
     }
 }
 
+std::optional<std::string> HTTPRequest::GetQueryParameter(const std::string& key) const
+{
+    const char* uri = evhttp_request_get_uri(req);
+
+    return GetQueryParameterFromUri(uri, key);
+}
+
+std::optional<std::string> GetQueryParameterFromUri(const char* uri, const std::string& key)
+{
+    evhttp_uri* uri_parsed = evhttp_uri_parse(uri);
+    const char* query = evhttp_uri_get_query(uri_parsed);
+    std::optional<std::string> result;
+
+    if (query) {
+        // Parse the query string into a key-value queue and iterate over it
+        struct evkeyvalq params_q;
```

Comment:
Is the `struct` keyword needed here?

```suggestion
       evkeyvalq params_q;
```

Reply:
This confused me too. I don't see how it's necessary to keep it, but that's how libevent [documents](https://libevent.org/doc/http_8h_source.html) it and also how it's already used everywhere ([1](https://github.com/bitcoin/bitcoin/blob/767ee2e3a1082468b4e2248bac3ef8bd54bb2ddb/src/httpserver.cpp#L525), [2](https://github.com/bitcoin/bitcoin/blob/767ee2e3a1082468b4e2248bac3ef8bd54bb2ddb/src/httpserver.cpp#L556), [3](https://github.com/bitcoin/bitcoin/blob/3c565302aaa91ae46aa09b8aeb95206711d9d8a6/src/bitcoin-cli.cpp#L766)) in Bitcoin Core, as well as in most (all?) unofficial examples/documentation you can find online. I thought it might be a good idea to keep it to stay in line with libevent conventions. Knowing that, would you still prefer to have it removed? I don't have a strong view either way, but think I prefer sticking with the convention.

Reply:
ok, I've dug into this a bit more, and it's an [elaborated type specifier](https://en.cppreference.com/w/cpp/language/elaborated_type_specifier). It means that even if we have a local variable named `evkeyvalq`, then this would still compile, since the `struct` class-key would mean the local variable is ignored.

I think it's fine to leave it.